### PR TITLE
Use tip to ensure each transaction's inclusion in the next block. 

### PR DESCRIPTION
**Pull request by cliffhall**
## What is this
Refactor the deployment script to use tip rather than gas price and/or limit to ensure a each transaction's inclusion in the block.

## Why 
* For details of the strategy, see https://docs.alchemy.com/docs/maxpriorityfeepergas-vs-maxfeepergas

* Suffice it to say, supplying only an attractive tip amount is the best way to ensure inclusion in the next block.

## What changed
* In env.example
  - set CONFIRMATIONS to 3 (we cannot run with only one confirmation on polygon with high uncle rate. Probably this needs to be between 5 and 10, we'll see

* In deploy-protocol-*.js,
  - replaced gasLimit with maxPriorityFeePerGas

* In deploy-suite.js,
  - fetch tipMultiplier from environment
  - set tipSuggestion to amount that is always returned from ethers
  - create maxPriorityFeePerGas as bignumber with tipMultiplier times tipSuggestion
  - remove fetch of gasLimit from environment, no longer needed (pre-1559)
  - in all deployment calls that passed gasLimit, now pass maxPriorityFeePerGas

* In environments.js,
  - fetch tipMultiplier from process.env.TIP_MULTIPLIER

### CODE REVIEW

**Review by zajck — DISMISSED** [review]
In `deploy-suite.js` we have few more calls to the contracts (`grantRole`, `setVoucherBeaconAddress`, `setBeaconProxyAddress`, `setAuthTokenContract`, `renounceRole`), which should then also use `maxPriorityFeePerGas`

File: scripts/deploy-suite.js [hunk 1]
```diff
@@ -2,8 +2,10 @@ const environments = require("../environments");
 const hre = require("hardhat");
 const ethers = hre.ethers;
 const network = hre.network.name;
-let gasLimit;
 const confirmations = environments.confirmations;
```
[the remark this answers is not in the document]
Comment (zajck) [reply scripts/deploy-suite.js]: I already made this change.

File: scripts/util/deploy-protocol-client-proxies.js [hunk 2]
```diff
@@ -19,18 +20,18 @@ const confirmations = environments.confirmations;
...
   // Destructure the protocol client implementations
   [bosonClientBeacon] = protocolClients;
 
   // Deploy the ClientProxy for BosonVoucher
   const ClientProxy = await ethers.getContractFactory("BeaconClientProxy");
-  const clientProxy = await ClientProxy.deploy({ gasLimit });
+  const clientProxy = await ClientProxy.deploy(await getFees(maxPriorityFeePerGas));
   await clientProxy.deployTransaction.wait(confirmations);
 
   // init instead of constructors
```
Comment (hswopeams) [inline scripts/util/deploy-protocol-client-proxies.js:37]: Do need this for the initialize call just below this on line 38?

**Review by cliffhall — COMMENTED** [review]
LGTM! 👍

**Review by hswopeams — COMMENTED** [review]
Just one question. See comment.

File: scripts/util/deploy-protocol-client-proxies.js (hunk 2 above)
Comment (zajck) [reply scripts/util/deploy-protocol-client-proxies.js:37]: Added.

**Review by hswopeams — APPROVED** [review]

**Review by dennisfurrer — APPROVED** [review]

**Pull request merged**